index.d.ts

```diff
@@ -28,5 +28,6 @@ declare function arrify(value: string): [string];
 declare function arrify<ValueType>(value: ValueType[]): ValueType[];
 declare function arrify<ValueType>(value: Iterable<ValueType>): ValueType[];
 declare function arrify<ValueType>(value: ValueType): [ValueType];
+declare function arrify<ValueType>(value: ValueType|ValueType[]): ValueType[];
```

Comment:
I think that we should rethink the whole definion. What you propose only fixes the case that the parameter is of a special type or an array of this type (`number | number[]`). What about `number | string[]` or any other type combination?

I've tried this one, it seems to work as expected but is very ugly:
```suggestion
declare function arrify<ValueType>(
	value: ValueType
): ValueType extends (null | undefined)
	? []
	: ValueType extends string
	? [string]
	: ValueType extends (infer T)[]
	? T[]
	: ValueType extends ReadonlyArray<infer T>
	? ReadonlyArray<T>
	: ValueType extends Iterable<infer T>
	? [T]
	: [ValueType];
```

Do you have a better idea?

Reply:
// @ajafff

Reply:
> : ValueType extends Iterable<infer T>
> ? [T]

`[T]` should be `T[]`

Reply:
Right, thank you!

Reply:
@callmehiphop Could you please update your PR with the suggested changes?

Reply:
I'm ok closing this one out if you want to open a new PR with those changes

Reply:
Ok, close this one, I'll open a new one.